Разметка главной страницы

VoblPi opened the pull request:
---
:mortar_board: [Разметка главной страницы](https://up.htmlacademy.ru/htmlcss/33/user/1872141/tasks/3)

Vital-Sh committed:
Добавляет header + main (частично)

Vital-Sh committed:
Разметка index.html

Скорее всего ошибок предостаточно...

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+          <li class="footer_nav_list"><a href="#">Доставка</a></li>
+          <li class="footer_nav_list"><a href="#">Гарантия</a></li>
+          <li class="footer_nav_list"><a href="#">Контакты</a></li>
+          </li>
```
Лишний закрывающий тег `</li>`.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
-
+    <header class="header-index">
+      <h1 class="logo"><a href="#"><img src="" alt="Логотип интернет-магазина Device"></a></h1>
```
`h1` лучше из логотипа не делать. Просто `<a href="#"><img src="" alt="Логотип интернет-магазина Device"></a>`. Но поскольку `h1` нужен на странице, то в `main` надо добавить скрытый заголовок.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+      <h1 class="logo"><a href="#"><img src="" alt="Логотип интернет-магазина Device"></a></h1>
+      <div class="header_user-menu">
+        <div class="header_search">
+          <input class="search_input" type="search" placeholder="Поиск по сайту">
+          <input class="search_button" type="button" name="search" value="Найти">
```
`type="submit"`, так как простая кнопка не отправит данные из формы. Однако тут лучше использовать не `input`, а `button type="submit"`, так как `input type="submit" устаревший элемент.

andreysgra replied:
Это элементы формы. Без формы, сами по себе, они не имеют особого смысла. По формам есть [тренажёр](https://up.htmlacademy.ru/htmlcss/33/module/2/item/6). Также можно посмотреть разметку формы в [этом демо](https://up.htmlacademy.ru/htmlcss/33/demos/6169#37).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+          <input class="search_input" type="search" placeholder="Поиск по сайту">
+          <input class="search_button" type="button" name="search" value="Найти">
+        </div>
+        <a class="login_link" href="#">Вход</a>
```
Здесь стоит использовать список ссылок, так как на странице каталога показана версия авторизованного пользователя:
![image](https://user-images.githubusercontent.com/17790771/152360614-1bf75035-e843-4875-8088-7191f211eba3.png)

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+          <input class="search_button" type="button" name="search" value="Найти">
+        </div>
+        <a class="login_link" href="#">Вход</a>
+        <nav class="navigation_user">
+          <ul class="navigation_user_list">
+            <li class="navigation_user_compare"><a href="#">Сравнить</a></li>
+            <li class="navigation_user_basket"><a href="#">Корзина</a></li>
+          </ul>
+        </nav>
```
Здесь не нужно использовать тег `<nav>` так как этот тег обозначает главную навигацию по сайту.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+            </div>
+            <div>
+              <a class="slider_item-quadrocopter_more" href="#">Подробнее</a>
+              <button class="slider_button slider_button-current" type="button">
+                <span class="visually-hidden">1 фото</span>
+              </button>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">2 фото</span>
+              </button>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">3 фото</span>
+              </button>
```
В слайдере (общий блок для всех трёх слайдов) есть блок кнопок для переключения слайдов:
![image](https://user-images.githubusercontent.com/17790771/150852639-1b19586d-8c91-4963-9cd1-d23e65369c4c.png)
Этот блок один на весь **слайдер**. В каждом слайде не нужно его дублировать (и он не должен присутствовать в теле слайда. **Слайдер** - это общий блок, который объединяет все слайды).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+              <span>Радиус поражения</span>
+            </div>
+          </li>
+          <li class="slider_item">
+            <div class="slider_item-next">
+              <a class="slider_item-next_more" href="#">Подробнее</a>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">1 фото</span>
+              </button>
+              <button class="slider_button slider_button-current" type="button">
+                <span class="visually-hidden">2 фото</span>
+              </button>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">3 фото</span>
+              </button>
+            </div>
+          </li>
+          <li class="slider_item">
+            <div class="slider_item-last">
+              <a class="slider_item-last_more" href="#">Подробнее</a>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">1 фото</span>
+              </button>
+              <button class="slider_button" type="button">
+                <span class="visually-hidden">2 фото</span>
+              </button>
+              <button class="slider_button slider_button-current" type="button">
+                <span class="visually-hidden">3 фото</span>
+              </button>
+            </div>
+          </li>
```
Если посмотреть страницу стайлгайда, то на фрейме **slider** можно заметить три слайда, а не один. Разметить нужно их все.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+    <main class="main-index">
+      <section class="main_slider">
+        <h2 class="visually-hidden">Слайдер с новинками интернет-магазина</h2>
+        <div class="slider_image">
+          <img class="slider_image-quadrocopter" src="" alt="Квадрокоптер с лазером, замаскированным под камеру" >
+          <button class="slider_button slider_prev" type="button">
+            <span class="visually-hidden">Предыдущее фото</span>
+          </button>
+          <button class="slider-button slider-next" type="button">
+            <span class="visually-hidden">Следующее фото</span>
+          </button>
+        </div>
+        <ol class="slider_properties">
+          <li class="slider_item">
+            <div class="slider_item-quadrocopter">
+              <p class="quadrocopter-promo">Порхает как бабочка, жалит как пчела!</p>
+              <p>Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером, замаскированным под стандартную камеру.</p>
+            </div>
+            <div>
```
Какой здесь должен быть подход к разметке:
- это тело одного слайда (всего слайдов 3)
![image](https://user-images.githubusercontent.com/17790771/150852088-1968cbc6-1a24-4aeb-bd81-879798398f0a.png)
- в слайде есть блок с фотографиями (фото представлено одно, поэтому достаточно разметить только его
![image](https://user-images.githubusercontent.com/17790771/150852287-143b89b6-2611-441d-816d-e92b77ca91aa.png)
- в слайде есть блок с описанием товара
![image](https://user-images.githubusercontent.com/17790771/150852357-5d8a5811-8f0f-4a63-817b-7df1274f77de.png)

В блоке с фотографиями есть кнопки для переключения фото:
![image](https://user-images.githubusercontent.com/17790771/150852466-a54837ac-2e4d-4187-b7b4-c9934ec9f4e0.png)
На весь блок одни кнопки, но в каждом слайде свой блок кнопок.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+      <section class="category_list">
+        <h2 class="visually-hidden">Категории товаров из каталога</h2>
+        <ul>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок очков виртуальной реальности"><span>Виртуальная реальность</span></a>
+          </li>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок монопода"><span>Моноподы для селфи</span></a>
+          </li>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок экшн-камеры"><span>Экшн-камеры</span></a>
+          </li>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок фитнес-браслета"><span>Фитнес-браслеты</span></a>
+          </li>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок часов"><span>Умные часы</span></a>
+          </li>
+          <li class="category_list_virtual-reality">
+            <a href="#"><img src="" alt="Схематичный рисунок квадрокоптера"><span>Квадрокоптеры</span></a>
+          </li>
```
Иконки в этом блоке
![image](https://user-images.githubusercontent.com/17790771/152362641-f5adeb0a-3591-45d9-9b62-60035a7d32e8.png)
являются декоративными изображениями, следовательно тег `img` под них использовать не нужно (изображения будут добавлены средствами CSS). [Как отличить контентное изображение от декоративного](https://htmlacademy.ru/blog/boost/frontend/content-or-decor-img).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+            <button class="benefits_button" type="button">Кредит</button>
+          </li>
+        </ul>
+        <span>Доставка</span>
+        <p>Мы с удовольствием доставим ваш товар прямо к вашему подъезду совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш этаж.</p>
```
Здесь, как и в случае с предыдущим слайдером, три слайда (смотрите страницу стайлгайда).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+      </section>
+      <a class="delivery" href="#">Доставим редкий товар под заказ</a>
+      <section class="about-shop">
+        <h2 class="visualy-hiden">Информация о магазине и контакты</h2>
+        <div class="about-shop_about-us">
+          <h2 class="about-us">О нас</h2>
+          <p>Огромный выбор гаджетов не оставит равнодушным гика, который есть в каждом из нас.</p>
+          <dl>
+            <dt>Мы можем доставить ваш товар в самые отдалённые уголки России транспортными компаниями:</dt>
+            <dd>Меловые линии</dd>
+            <dd>СДИК</dd>
+            <dd>ТоксБерри</dd>
+          </dl>
+          <a href="">Подробнее о нас</a>
+        </div>
+        <div class="about-shop_contacts">
+          <h2 class="contacts">Контакты</h2>
```
Здесь нарушается иерархия заголовков. У родительского элемента, секции, уже есть заголовок второго уровня _"Информация о магазине и контакты"_, следовательно во вложенных блоках заголовки должны быть третьего уровня.

andreysgra commented on index.html:
```diff
@@ -5,8 +5,214 @@
+        <div class="about-shop_contacts">
+          <h2 class="contacts">Контакты</h2>
+          <p>Вы можете забрать товар сами, заехав в наш офис. Заодно сможете проверить работоспособность покупки.</p>
+          <h3 class="contacts-subtitle">Интернет-магазин «Девайс»</h3>
```
А это будет уже заголовок четвёртого уровня.

andreysgra reviewed (changes requested):
Несмотря на определённые замечания, разметка страницы выполнена хорошо.

Vital-Sh committed:
Исправил замечание

Vital-Sh committed:
Провёл работа над ошибками

Увидел, что ошибок больше чем одна и исправил

andreysgra commented on index.html:
```diff
@@ -5,8 +5,227 @@
+          </form>
+        </li>
+        <li><a class="login_link" href="#">Вход</a></li>
+        <div class="navigation_user">
+          <li class="navigation_user_compare"><a href="#">Сравнить</a></li>
+          <li class="navigation_user_basket"><a href="#">Корзина</a></li>
+        </div>
```
`div` не может быть дочерним элементом списка. Разметку желательно проверять в [валидаторе](https://validator.w3.org/nu/#textarea) перед отправкой задания на проверку.

VoblPi replied:
Не заметил, что ошибся тут... Вместо div тут ul...

andreysgra commented on index.html:
```diff
@@ -5,8 +5,227 @@
-
+    <header class="header_index">
+      <a class="logo" href="#"><img src="" alt="Логотип интернет-магазина Device"></a>
+      <ul class="header_user_menu">
+        <li class="header_search">
+          <form class="header_search_container" action="https://echo.htmlacademy.ru/">
+            <input type="text" id="search_input" placeholder="Поиск по сайту">
+            <button class="search_button" type="submit">Найти</button>
+          </form>
+        </li>
+        <li><a class="login_link" href="#">Вход</a></li>
+        <div class="navigation_user">
+          <li class="navigation_user_compare"><a href="#">Сравнить</a></li>
+          <li class="navigation_user_basket"><a href="#">Корзина</a></li>
+        </div>
+      </ul>
```
Предыдущий вариант разметки у вас был вполне логичным и не совсем понятно, зачем вы объединили элементы разного назначения в один список. Список должен объединять схожие по смыслу сущности.

**_Логичный список._**
Список дел:
- проснуться в семь утра;
- умыться;
- позавтракать;
- пойти на работу.

**_Нелогичный список._**
Список дел:
- проснуться в семь утра;
- зелёный;
- лето;
- компьютер.

Это [моё замечание](https://github.com/htmlacademy-htmlcss/1872141-device-33/pull/2#discussion_r798609857)  относилось к ссылке "Вход" и я рекомендовал использовать здесь не просто ссылку, а список ссылок (на главной странице список с одним элементом, потому что показана страница неавторизованного пользователя, а на странице каталога список из двух элементов - авторизованный пользователь).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,227 @@
+      <ul class="header_user_menu">
+        <li class="header_search">
+          <form class="header_search_container" action="https://echo.htmlacademy.ru/">
+            <input type="text" id="search_input" placeholder="Поиск по сайту">
```
Для `input` лучше использовать `type="search"` (в прошлом варианте у вас так и было).

andreysgra commented on index.html:
```diff
@@ -5,8 +5,227 @@
+      <h1 class="visually-hidden">Интернет-магазина Device</h1>
+      <section class="main_slider">
+        <h2 class="visually-hidden">Слайдер с новинками интернет-магазина</h2>
+        <div class="slider_button_arrows">
+          <button class="slider_button slider_prev" type="button">
+            <span class="visually-hidden">Предыдущее фото</span>
+          </button>
+          <button class="slider_button slider-next" type="button">
+            <span class="visually-hidden">Следующее фото</span>
+          </button>
+        </div>
```
Не совсем верно - данный блок переключения фотографий должен присутствовать в каждом слайде, так как на один товар (слайд) может быть несколько фото и слайды не связаны друг с другом никак.

Здесь этот  блок уберите и продублируйте его в каждом элементе `slider_item`.

Vital-Sh committed:
Исправил замечания 2

По макету не ясно "Вход" - это ссылка, или кнопка, открывающая форму для ввода логина и пароля.  Поэтому не могу понять в какой контейнер его записать, чтобы потом выравнивать... Так же не понял, почему в слайдере блок стрелок переключения слайдов "предыдущий - следующий" должен дублироваться для каждого слайда. Он же одинаков для всех слайдов, почему тогда его нельзя разметить в теле слайдера?
